### PR TITLE
[REVIEW]:Add groupProperties parameter for UA_ReaderGroupConfig

### PR DESCRIPTION
**Pull request by opcua-tsn-team-kalycito**

### CODE REVIEW

**Comment by jpfr** [comment]
Looks good.
Does this have to be an array of KeyValuePair or could it be a linked list of parameters?

For the latter we already have tooling place that can be easily reused.
https://github.com/open62541/open62541/blob/master/include/open62541/util.h#L40

**Comment by opcua-tsn-team-kalycito** [comment]
As per specification (Part-14) we need to have the GroupProperties parameter as an array of DataType KeyValuePair, refer below,

`6.2.4.6 GroupProperties The GroupProperties parameter is an array of DataType KeyValuePair that specifies additional properties for the configured group.`


> Looks good.
> Does this have to be an array of KeyValuePair or could it be a linked list of parameters?
> 
> For the latter we already have tooling place that can be easily reused.
> https://github.com/open62541/open62541/blob/master/include/open62541/util.h#L40

File: src/pubsub/ua_pubsub_reader.c [hunk 1]
```diff
@@ -481,11 +487,22 @@ UA_Server_ReaderGroup_clear(UA_Server* server, UA_ReaderGroup *readerGroup) {
...
                           UA_ReaderGroupConfig *dst) {
+    UA_StatusCode retVal = UA_STATUSCODE_GOOD;
     /* Currently simple memcpy only */
     memcpy(dst, src, sizeof(UA_ReaderGroupConfig));
     memcpy(&dst->securityParameters, &src->securityParameters, sizeof(UA_PubSubSecurityParameters));
-    UA_String_copy(&src->name, &dst->name);
-    return UA_STATUSCODE_GOOD;
+    retVal |= UA_String_copy(&src->name, &dst->name);
+    if (src->groupPropertiesSize > 0) {
+        dst->groupPropertiesSize = src->groupPropertiesSize;
```
Comment (jpfr) [inline src/pubsub/ua_pubsub_reader.c:496]: Set the size after checking for null.

**Review by jpfr — APPROVED** [review]
One last small thing to correct.
Will be merged then.

**Comment by jpfr** [comment]
Replaced by #4506.

**Pull request closed without merging**